audformat/core/database.py

```diff
@@ -558,6 +565,21 @@ def join_field(
                 else:
                     self[table_id] = table.copy()
 
+        # copy media files
+        if copy_media and self.root:
+            for other in others:
+                if len(other.files) > 0 and other.root is None:
+                    raise RuntimeError(
+                        f"The database '{other.name}' has no root folder."
+                    )
+                for file in other.files:
+                    # Joining works also if the paths are already absolute
```

Comment:
Yes, but then `dst_file` points to `src_file` :)

Reply:
No, why should this happen.

The comment refers to:

```python
>>> os.path.join('/a', '/a/b')
'/a/b'
```

Reply:
`dst_file` and `src_file` should be different as `self.root` and `other.root` should be different.

Reply:
yes, but the comment suggests that paths can be absolute, but that is not true, they have to be relative.

Reply:
No `/a/b` in my example is absolute and it works. If you load a database with `audb.load()` you can get absolute paths.

Reply:
Ok, I'm confused.

```
file = '/a/b/file.wav'
dst_file = os.path.join(self.root, file)  # ->  '/a/b/file.wav'
src_file = os.path.join(other.root, file) # -> '/a/b/file.wav'
```

No?

Reply:
Ah, ok, now I see the problem.

I have to fix it for `dst_file`, will have a look.

Reply:
Something like:

```python
if file.startswith(other.root):
    file = file.replace(other.root, '')
```

Reply:
I fixed the absolute path by removing it from `file` and added a test for it.

Reply:
But I guess there is still the problem that the new entries in the tables are still pointing to the old location?

Reply:
That's true and it gets even more complicated:

* if `self.db` stores the tables absolute, we have to replace the old absolute path with the new absolute path, or the add the new `db.root` path if other was relative
* if `self.db` stores the tables relative, we have to remove the absolute path

The other think is, we cannot make those changes before by applying it on `other` as this would change the original `other`  database which we don't want. So we first need to join and afterwards fix all the tables in `self`.

Reply:
Oh and it's even worse - with absolute paths, we get duplicated entries for every file that exists in both databases :) So maybe it is easier if we simply forbid absolute paths in `update()`?

Reply:
hehe, kind of same thought at the same time :)

Reply:
Yes, for a start it seems the easiest solution, I will adapt the code and forbid absolute path in `others` and `self`.

Reply:
OK, I'm struggling to get it to work.
I will pause here and continue on Monday.

Reply:
Just a remark on what is not working yet:

The test if the file is absolute in `other` happens to late at the moment, we have to fail before `copy_media=True` because the tables are joined already:

```
Index(['/tmp/pytest-of-hwierstorf/pytest-125/test_update0/other2/audio/001.wav',
       '/tmp/pytest-of-hwierstorf/pytest-125/test_update0/other2/audio/002.wav',
       '/tmp/pytest-of-hwierstorf/pytest-125/test_update0/other2/audio/003.wav',
       '/tmp/pytest-of-hwierstorf/pytest-125/test_update0/other2/audio/004.wav',
       '/tmp/pytest-of-hwierstorf/pytest-125/test_update0/other2/audio/005.wav',
       'audio/000.wav', 'audio/001.wav', 'audio/002.wav', 'audio/003.wav',
       'audio/004.wav', 'audio/005.wav'],
      dtype='object', name='file')
```

Reply:
By don't you move the test to the front of the function?

Reply:
And the test should of course be applied to `self` and `other`.

Reply:
And it should be applied independently of `copy_media`.

Reply:
I moved the checks for absolute path now to the beginning of `update()` and it works.